### DISCUSSION TITLE
Use sql for a complex columnTransformer?

### DISCUSSION THREAD

**Discussion by sqcsabbey** [simolus3/drift#2395] [Q&A]
I want to add a non-null IntColumn to my database, and I need it populated with random numbers from 1-999. I tried the below...

```
          await m.alterTable(TableMigration(
            points,
            newColumns: [points.nameIndex],
            columnTransformer: {
              points.nameIndex:
                  // random 1-999
                  const Constant('abs(random() % 999) + 1'),
            },
          ));
```

... which failed with:

```
I/flutter (25062): ⚠️ Unhandled Dart error: FormatException: Invalid radix-10 number (at character 1)
I/flutter (25062): ⚠️ abs(random() % 999) + 1
```

Is there a way to pass an expression like this? I was also trying something similar on a new non-nullable TextColumn, and the Constant was using the literal string for my sql (`Constant('char(abs(random() % 26) + 65)')`); this is a complex expression to represent with `FunctionCallExpression` (I tried starting down that path...).

**Comment by simolus3** [comment]
This sounds like a job for [`CustomExpression`](https://drift.simonbinder.eu/api/drift/customexpression-class), which should make the column transformer work.
